### ISSUE TITLE
PyTorch VoxelMorph: registrations to obtain segmentation -> segmentation masks are not smooth.

### ISSUE THREAD

**Issue by BraveDistribution** [voxelmorph/voxelmorph#522]
**Task** (what are you trying to do/register?)

I am working on US 2D cardiac cycle registration, with the goal of obtaining masks for all images in the cycle by using mask warping with the deformation field.

**What have you tried**

I have tried different hyperparameters, image sizes, and training methods (e.g., sequential registration of t0->t1, t1->t2, or one-to-all such as t0->t3, t0->t5). From my experiments, I found that sequential registration works better.

**Details of experiments**

I am using the version from this page (pytorch one). The regularization loss (grad loss) weight is set to 0.01 as suggested.  

I am not allowed to share the input and output data, however the masks could be doable:
t0: 
![image](https://user-images.githubusercontent.com/36101458/233982926-aee89268-f62b-4ddc-9f77-0cc149a6b592.png)

t1: 
![image](https://user-images.githubusercontent.com/36101458/233982959-35e57e99-16c7-44c3-94e1-ef02a77d0a45.png)

t2: ... so to the end of the cycle (shrinkage of the mask) it works well. However, when doing upsampling of the mask, it starts to crumble.

tx: The masks look good for the beginning of the cycle but start to crumble when doing upsampling, as shown in the images for tx. I suspect this may be due to the interpolator (spatial transformer) not being smooth enough, even though I am using the 'nearest' mode. As we work in a sequential way, the smoothness gradually worsens.

![image](https://user-images.githubusercontent.com/36101458/233983141-7ecd6b8e-347f-4366-a007-923b0c4d754f.png)

![image](https://user-images.githubusercontent.com/36101458/233983324-694e38f5-6806-4baf-b6c7-a6e3a0431bc2.png)

any tips how to make the interpolator (spatialtransformer) more smooth? I use 'nearest' in the mode.

The registered image looks okay and also the training losses:
![image](https://user-images.githubusercontent.com/36101458/233983962-4092e817-ed2f-44bd-a4ae-7cf9c6fb860e.png)

These are the parameters:
![image](https://user-images.githubusercontent.com/36101458/233983683-291e9f22-0c99-4490-b6c0-8e2ccc5148b6.png)

I am doing only 1 transformation (scaleintensity from torchvision). 

If you have any tips to make the interpolator smoother, I would greatly appreciate it. The data I am working with is similar to the one in this GitHub issue. Thank you!
